plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionStreamingAction.java

```diff
@@ -0,0 +1,271 @@
+                                    throw new MLException("Sending http chunks failed.");
+                                }
+                            }
+
+                            @Override
+                            public void onFailure(Exception e) {
+                                throw new MLException("Got an exception in MLPredictionTaskAction.", e);
+                            }
+                        });
+                    } catch (IOException e) {
+                        throw new MLException("Got an exception in flux.", e);
+                    }
+                })
+                .onErrorComplete(ex -> {
+                    if (ex instanceof Error) {
+                        log.info("Got an error in flux");
+                        return false;
+                    }
+                    try {
+                        channel.sendResponse(new BytesRestResponse(channel, (Exception) ex));
+                        return true;
+                    } catch (final IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                })
+                .subscribe();
+        };
+
+        return channel -> {
+            if (channel instanceof StreamingRestChannel) {
+                consumer.accept((StreamingRestChannel) channel);
+            } else {
+                final ActionRequestValidationException validationError = new ActionRequestValidationException();
+                validationError.addValidationError("Unable to initiate request / response streaming over non-streaming channel");
+                channel.sendResponse(new BytesRestResponse(channel, validationError));
+            }
+        };
+    }
+
+    @Override
+    public boolean supportsContentStream() {
+        return true;
+    }
+
+    @Override
+    public boolean supportsStreaming() {
+        return true;
+    }
+
+    @Override
+    public boolean allowsUnsafeBuffers() {
+        return true;
+    }
+
+    /**
+     * Creates a MLPredictionTaskRequest from a RestRequest
+     *
+     * @param request RestRequest
+     * @return MLPredictionTaskRequest
+     */
+    @VisibleForTesting
+    MLPredictionTaskRequest getRequest(String modelId, String algorithm, RestRequest request, BytesReference content) throws IOException {
+        ActionType actionType = ActionType.from(getActionTypeFromRestRequest(request));
+        if (FunctionName.REMOTE.name().equals(algorithm) && !mlFeatureEnabledSetting.isRemoteInferenceEnabled()) {
+            throw new IllegalStateException(REMOTE_INFERENCE_DISABLED_ERR_MSG);
+        } else if (FunctionName.isDLModel(FunctionName.from(algorithm.toUpperCase(Locale.ROOT)))
+            && !mlFeatureEnabledSetting.isLocalModelEnabled()) {
+            throw new IllegalStateException(LOCAL_MODEL_DISABLED_ERR_MSG);
+        } else if (ActionType.BATCH_PREDICT == actionType && !mlFeatureEnabledSetting.isOfflineBatchInferenceEnabled()) {
```

Comment:
Do we support stream for `BATCH_PREDICT` ? If no, I think we should throw exception directly, no need to check `mlFeatureEnabledSetting.isOfflineBatchInferenceEnabled()`